src/lib/registration/connect_helpers.rb

```diff
@@ -103,9 +103,8 @@ def self.catch_registration_errors(message_prefix: "",
           check_smt_api(error_msg)
           report_error(message_prefix + _("Connection to registration server failed."), error_msg)
         when 422
-          if silent_reg_code_mismatch && e.response.body["error"] =~
-              /does not include the requested product/
-            log.info "Reg code does not work for this product, that's OK"
+          if silent_reg_code_mismatch
+            log.info "Reg code does not work for this product"
```

Comment:
We should log `e.response.body["error"]` so that we can debug things the next time the API changes under our hands.

Reply:
Is not enough with the [`log.error`](https://github.com/yast/yast-registration/blob/c39310276f37415981a883dfab64cc977a1f578c/src/lib/registration/connect_helpers.rb#L95) at the beginning?

https://github.com/yast/yast-registration/blob/c39310276f37415981a883dfab64cc977a1f578c/src/lib/registration/connect_helpers.rb#L94-L97

I will add a simple test example for it.

Reply:
Oh, you are right, I did not look that far up :)